engine/src/main/java/org/hibernate/search/backend/impl/DeleteByQuerySupport.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.NumericUtils;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.backend.SingularTermQuery;
+import org.hibernate.search.backend.SingularTermQuery.Type;
+import org.hibernate.search.exception.AssertionFailure;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * This class has means to convert all (by default) supported DeletionQueries back to Lucene Queries and to their
+ * String[] representation and back.
+ *
+ * @author Martin Braun
+ */
+public final class DeleteByQuerySupport {
+
+	private DeleteByQuerySupport() {
+		throw new AssertionError( "can't touch this!" );
+	}
+
+	public static final Map<Integer, ToLuceneQuery> TO_LUCENE_QUERY_CONVERTER;
```

Comment:
I'm wondering here if we really want to keep the Map.
I understand it's an implementation level class, and that sometimes it's convenient to expose public fields. But being a public Map with an `Integer` key, that's not a nice contract for clients (even if it's impl).
I'd rather expose a better API, and internally use a more efficient array. That's if we really need to use the integer based key.

Looking at the different parse functions, it looks like you have to validate for a specific lenght of the array. Maybe a fully typesafe API is better?
You'd be able to stricly mandate the expected parameters at compile time.